components/LobbyCircle.tsx

```diff
@@ -15,18 +20,58 @@ const CreateOptions2Style = {
 
 export interface LobbyCircleProps {
   name: string;
+  href?: string;
+  onClick?: () => void;
   create?: CreateOptions;
 }
 
-export function LobbyCircle({name, create = CreateOptions.NotCreate}: LobbyCircleProps): JSX.Element {
-  return (
-    <a href='/' className={`${styles.circle} ${CreateOptions2Style[create]}`}>
-      <div className={styles.name}>
-        {name}
-      </div>
-    </a>
+export function LobbyCircle({
+  name,
+  href,
+  onClick = () => null,
+  create = CreateOptions.NotCreate,
+}: LobbyCircleProps): JSX.Element {
+  const ref = useRef(null);
+  const circRef = useRef(null);
+  const tlRef = useRef(null);
+  const [animateForwards, animateBackwards] = useTimelineControls({tlRef, animate: true});
+
+  useEffect(() => {
+    tlRef.current = makeRotationTimeline(circRef.current);
+  }, []);
+
+  useEffect(() => {
+    const circText = new CircleType(ref.current);
+
+    function setRadius() {
+      circText.radius(window.innerHeight * .35 / 2).dir(-1);
+    }
+    setRadius();
+    window.addEventListener('resize', setRadius);
 
+    return () => {
+      window.removeEventListener('resize', setRadius);
+      circText.destroy();
+    };
+  }, [ref]);
+
+  const padName = name?.replace(/./g, '$& ');
+
+  const base = (
+    <div
+      className={`${styles.circle} ${CreateOptions2Style[create]}`}
+      onClick={onClick}
+      ref={circRef}
+      onMouseEnter={animateForwards}
+      onMouseLeave={animateBackwards}
+    >
+      <PlaylistVisual title='' subtitle='' fullSize={false} animate={false} />
+      <div className={styles.name} ref={ref}>
+        {padName}
+      </div>
+    </div>
   );
+  return href ? (<NextLink href={href}>{base}</NextLink>) : base;
```

Comment:
Let's modify the `Link` component  that i already made to cover this feature too.. maybe like a `isDisabled` prop